### PR TITLE
Add more Spanner Tests

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: spanner/api/Program.cs [hunk 1]
```diff
@@ -695,6 +706,32 @@ await connection.OpenAsReadOnlyAsync()
             // [END query_data_with_new_column]
         }
 
+        public static async Task ListDatabaseTablesAsync(
+    string projectId, string instanceId, string databaseId)
```
Comment (SurferJeffAtGoogle) [inline spanner/api/Program.cs]: Fix indentation.  Have you run Lint-Code on this PR?

File: spanner/api/SpannerTest/Tests.cs [hunk 2]
```diff
@@ -121,7 +122,14 @@ void InitializeDatabase()
                 {
                     cmd.Parameters["SingerId"].Value = i;
                     cmd.Parameters["AlbumId"].Value = i;
-                    cmd.Parameters["MarketingBudget"].Value = 300000;
+                    if (i == 1)
```
Comment (SurferJeffAtGoogle) [inline spanner/api/SpannerTest/Tests.cs]:
```
cmd.Parameters["MarketingBudget"].Value = i == 1 ? firstAlbumBudget : secondAlbumBudget;
```

File: spanner/api/SpannerTest/Tests.cs [hunk 3]
```diff
@@ -222,5 +284,23 @@ static bool ContainsError(AggregateException e, ErrorCode errorCode)
...
+        /// </summary>
+        /// <param name="e">The exception to examine.</param>
+        /// <param name="errorCode">The error code to look for.</param>
+        /// <returns></returns>
+        static bool ContainsGrpcError(AggregateException e, Grpc.Core.StatusCode errorCode)
+        {
+            foreach (var innerException in e.InnerExceptions)
+            {
+                Grpc.Core.RpcException grpcException = innerException as Grpc.Core.RpcException;
+                if (grpcException != null && Grpc.Core.StatusCode.AlreadyExists == errorCode)
```
Comment (SurferJeffAtGoogle) [inline spanner/api/SpannerTest/Tests.cs]:
```
if (grpcException != null && grpcException.ErrorCode == errorCode)
```

**Comment by SurferJeffAtGoogle** [comment]
I see a second request for a review, but no new commits.

**Comment by jsimonweb** [comment]
Addressed review comments. Please review.

**Pull request merged**